### DISCUSSION TITLE
Persisting Data in a next.js ssr application

### DISCUSSION THREAD

**Discussion by Julius-R** [pmndrs/zustand#516] [Q&A] [answered]
Hello, I'm just getting into learning about zustand. I was having difficulty finiding resources on the topic, but I am wanting to know, how would I go about persisting data with zustand between pages in a next application that uses server side rendering?

**Comment by AnatoleLucet** [comment, marked as answer]
Hi, apart from what was discussed in #510 (which is pretty minor as long as you don't want to access the storage's data on the server), you should be able to use the [`persist` middleware](https://github.com/pmndrs/zustand#persist-middleware) as if it was any regular CSR React app.

**Reply by Julius-R** [reply]
Thanks. I looked over it, and this works!
